### ISSUE TITLE
Possible to add support for org headings?

### ISSUE THREAD

**Issue by sooheon** [angelic-sedition/emacs-sentence-navigation#3]
I think handling of `*` in org can be improved. For example, with:

```
* Heading 1
Foo bar baz. Quux.

* Heading 2
```

`das` from `Quux` will delete the leading `*` of the next heading. `das` from `Foo` will delete text of Heading 1.

Org headings (prefixed by `*`) have the property that they will never hard wrap, even when manually calling `M-q`. So it's safe to assume that the newline of an org heading line is always the end of a sentence. Can we support this so that something matching the org heading regex is always a sentence in and of itself?

Also, what would it take to implement the impossible case in the readme for lines ending with period and next sentence beginning immediately in hard-wrapped buffers?

**Comment by noctuid** [comment]
Yeah this package could use some work. Right now it makes some dumb assumptions about the start of a sentence (e.g. it won't work with a sentence starting at the beginning of the buffer). For text that isn't hard-wrapped, it's easiest to just assume that a capital letter (or a left quote or something) at the beginning of the line always starts a sentence. For it to work with hard-wrapped text as well, I'll need to add a variable for regexps that can precede the start of a sentence for different modes. In org mode this would also include things like line comments and tables I guess. I don't really use hard-wrapping most of the time, so I never did this, but I'll fix this problem next time I get the chance.

**Comment by noctuid** [comment]
Okay this example should now work as expected. I've made a lot of other improvements, so the behavior should be saner in general. Accurate support for hard-wrapped sentences is implemented and requires that `sentence-nav-hard-wrapping` is non-nil and that `sentence-nav-non-sentence-line-alist` is customized properly for the current major mode (which is fairly simple). I've added basic support for org mode. I'll flesh out the list of regexps for org mode and add support for markdown mode when I get the chance.
